File: frontend/scripts/create-icons.js
/**
 * Simple script to create placeholder PWA icons
 * Run with: node scripts/create-icons.js
 */

const fs = require('fs');
const path = require('path');

// Simple PNG creation using a basic approach
// Since we can't use ImageMagick, we'll create a minimal valid PNG

function createSimplePNG(size, outputPath) {
  // Create a simple green square PNG
  // This is a minimal valid PNG file (1x1 pixel, green)
  // For a proper icon, you'd want to use a library like 'sharp' or 'canvas'
  // But for now, we'll create a placeholder that at least won't break
  
  // Minimal PNG header for a 1x1 green pixel
  // In production, you should replace these with proper icons
  const pngHeader = Buffer.from([
    0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A, // PNG signature
    0x00, 0x00, 0x00, 0x0D, 0x49, 0x48, 0x44, 0x52, // IHDR chunk
    0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01, // 1x1 dimensions
    0x08, 0x02, 0x00, 0x00, 0x00, 0x90, 0x77, 0x53, 0xDE, // Bit depth, color type, etc.
    0x00, 0x00, 0x00, 0x0C, 0x49, 0x44, 0x41, 0x54, // IDAT chunk
    0x08, 0x99, 0x01, 0x01, 0x00, 0x00, 0x00, 0xFF, 0xFF, 0x00, 0x00, 0x00, 0x02, 0x00, 0x01, // Green pixel data
    0x00, 0x00, 0x00, 0x00, 0x49, 0x45, 0x4E, 0x44, 0xAE, 0x42, 0x60, 0x82 // IEND chunk
  ]);

  // For now, we'll create a note file instead
  // The user should replace these with proper icons
  const note = `# Placeholder Icon

This is a placeholder for ${size}x${size} icon.

To create proper icons:
1. Use the SVG template: icon.svg
2. Convert to PNG at ${size}x${size} using:
   - Online tool: https://realfavicongenerator.net/
   - Image editor (GIMP, Photoshop, etc.)
   - Or install sharp: npm install sharp
   - Then run: node -e "const sharp = require('sharp'); sharp('icon.svg').resize(${size}, ${size}).toFile('icon-${size}x${size}.png')"

For now, the app will work but icons won't display properly.
Replace this file with a proper PNG icon.
`;

  fs.writeFileSync(outputPath, note);
  console.log(`Created placeholder note at: ${outputPath}`);
  console.log(`⚠️  Please replace with a proper ${size}x${size} PNG icon`);
}

// Create icons directory if it doesn't exist
const publicDir = path.join(__dirname, '../public');
if (!fs.existsSync(publicDir)) {
  fs.mkdirSync(publicDir, { recursive: true });
}

// Create placeholder files
createSimplePNG(192, path.join(publicDir, 'icon-192x192.png'));
createSimplePNG(512, path.join(publicDir, 'icon-512x512.png'));

console.log('\n✅ Placeholder icon files created!');
console.log('📝 Next: Replace these with proper PNG icons (see PWA_ICONS_README.md)');

File: frontend/scripts/create-icons.py
#!/usr/bin/env python3
"""
Simple script to create placeholder PWA icons
Run with: python3 scripts/create-icons.py
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_icon(size, output_path):
    """Create a simple green icon with 'MP' text"""
    # Create a green square image
    img = Image.new('RGB', (size, size), color='#16a34a')
    draw = ImageDraw.Draw(img)
    
    # Try to use a font, fallback to default if not available
    try:
        # Try to use a system font
        font_size = size // 4
        font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size)
    except:
        try:
            font = ImageFont.load_default()
        except:
            font = None
    
    # Draw 'MP' text in the center
    text = "MP"
    if font:
        # Get text bounding box
        bbox = draw.textbbox((0, 0), text, font=font)
        text_width = bbox[2] - bbox[0]
        text_height = bbox[3] - bbox[1]
    else:
        # Estimate size if no font
        text_width = size // 3
        text_height = size // 4
    
    # Center the text
    x = (size - text_width) // 2
    y = (size - text_height) // 2
    
    draw.text((x, y), text, fill='white', font=font)
    
    # Save as PNG
    img.save(output_path, 'PNG')
    print(f"✅ Created {size}x{size} icon: {output_path}")

if __name__ == '__main__':
    try:
        from PIL import Image, ImageDraw, ImageFont
    except ImportError:
        print("❌ PIL (Pillow) not installed. Installing...")
        print("Run: pip3 install Pillow")
        print("\nOr use an online tool to create icons:")
        print("1. Go to https://realfavicongenerator.net/")
        print("2. Upload icon.svg")
        print("3. Download the generated icons")
        print("4. Place in frontend/public/")
        exit(1)
    
    # Create public directory if it doesn't exist
    public_dir = os.path.join(os.path.dirname(__file__), '../public')
    os.makedirs(public_dir, exist_ok=True)
    
    # Create icons
    create_icon(192, os.path.join(public_dir, 'icon-192x192.png'))
    create_icon(512, os.path.join(public_dir, 'icon-512x512.png'))
    
    print("\n✅ All icons created successfully!")
